### ISSUE TITLE
`indent` or `indent-binary-ops`: Inconsistent behaviour when combining with parenthesis

### ISSUE THREAD

**Issue by RioShopfront** [eslint-stylistic/eslint-stylistic#1009]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Depending on the operator being used the combination of binary operators and parenthesis will result in incorrect warnings.

With both `indent` and `indent-binary-ops` enabled, the below code produces incorrect or unwanted warnings.
```typescript
const x = 10;
const y = 4;
const z = 2;

return x +
    z + (
        x ** 2 + // error here
        y ** 3
    ); // error here
```

Where as this will not.
```typescript
const x = 10;
const y = 4;
const z = 2;

return x +
    z * (
        x ** 2 +
        y ** 3
    );
```

Likewise the below will produce errors, ideally this is the code we would like to write and enforce using stylistic
```typescript
return typeof data.requestId === "string" &&
    typeof data.success === "boolean" && (
        typeof data.message === "string" ||
        typeof data.message === "undefined"
    );
```
But this will not
```typescript
return typeof data.requestId === "string" ||
    typeof data.success === "boolean" && (
        typeof data.message === "string" ||
        typeof data.message === "undefined"
    );
```

### Reproduction

https://github.com/eslint-stylistic/eslint-stylistic/pull/1008

### Contributes

- [x] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests
